File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b_think/20260827_171807/typescript/004_json_filter/solution.ts
import * as fs from 'fs';

const data: { name: string; age: number; active: boolean; score: number }[] =
  JSON.parse(fs.readFileSync('input/data.json', 'utf-8'));

const result = data
  .filter((r) => r.active === true && r.age >= 30)
  .sort((a, b) => a.name.localeCompare(b.name));

process.stdout.write(JSON.stringify(result, null, 2) + '\n');

File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b_think/20260827_171807/python/004_json_filter/solution.py
import json

with open("input/data.json", "r") as f:
    data = json.load(f)

result = [
    record for record in data
    if record["active"] is True and record["age"] >= 30
]
result.sort(key=lambda r: r["name"])

print(json.dumps(result))